backend/src/main/java/com/bakdata/conquery/models/jobs/ImportJob.java

```diff
@@ -113,10 +88,20 @@ public static ImportJob create(
 			header.assertMatch(table);
 
 			final ImportId importId = new ImportId(table.getId(), header.getName());
+			Import processedImport = namespace.getStorage().getImport(importId);
 
-			if (namespace.getStorage().getImport(importId) != null) {
+			if (update) {
+				if (processedImport == null) {
+					throw new WebApplicationException(String.format("Import[%s] is not present.", importId), Response.Status.NOT_FOUND);
+				}
+				// before updating the import, make sure that all workers removed the last import
+				namespace.sendToAll(new RemoveImportJob(processedImport));
```

Comment:
Bin mir nicht sicher, aber ich glaube du solltest darüber noch `namespace.getStorage().removeImport(imp.getId());` aufrufen (so wie in `com.bakdata.conquery.resources.admin.rest.AdminDatasetProcessor#deleteImport`). Vermute, dass es im Prinzip kein Problem ist aber du hast dann theoretisch eine Zeit, wo es keine Daten zum Import gibt aber den Import.

Reply:
So hatte ich vorhin gemacht, jedoch meinte @thoniTUB dass `namespace.sendToAll(new RemoveImportJob(processedImport))` schon machen sollte und ausreichend sei

Reply:
@awildturtok so wie ich das gesehen habe kann auf dem Managernode über all `update` auf dem Storage ausgeführt werden. auf den Shards geht das noch nicht. Dafür müssten die Nachrichten noch erweitert werden.

Reply:
Ich denke nicht, dass das wichtig ist, da die Nachrichten garantiert sind in Reihenfolge verarbeitet zu werden.